server/schedulers/hot_region.go

```diff
@@ -633,22 +494,41 @@ func (bs *balanceSolver) tryAddPendingInfluence() bool {
 	switch {
 	case bs.rwTy == write && bs.opTy == transferLeader:
 		maxZombieDur = bs.sche.conf.GetRegionsStatZombieDuration()
+	case bs.rwTy == write && bs.opTy == movePeer:
+		if bs.best.srcDetail.Info.IsTiFlash || bs.best.dstDetail.Info.IsTiFlash {
```

Comment:
If the source and target have a different type, we should directly return?

Reply:
Not in the scope of this PR.